spec/infinispan_local_spec.js

```diff
@@ -183,6 +201,101 @@ describe('Infinispan local client', function() {
+                      'modify', function(key) { console.log('[Event] Modified key: ' + key); },
+                      {listenerId: 'blblbl'});
+
+                  var clientAddListenerRemove = client.addListener(
+                      'remove', function(key) { console.log('[Event] Removed key: ' + key); },
+                      {listenerId: listenerId});
+
+                  return Promise.all([clientAddListenerModify, clientAddListenerRemove]);
+              });
+
+          var clientCreate = clientAddListeners.then(
+              function() { return client.putIfAbsent('eventful', 'v0'); });
+
+          var clientModify = clientCreate.then(
+              function() { return client.replace('eventful', 'v1'); });
+
+          var clientRemove = clientModify.then(
+              function() { return client.remove('eventful'); });
+
+          var clientRemoveListener =
+              Promise.all([clientAddListenerCreate, clientRemove]).then(
+                  function(values) {
+                      var listenerId = values[0];
+                      return client.removeListener(listenerId);
+                  });
+
+      }).catch(t.failed(done)).finally(done);
+  });
+  it('can handle the case when listener fails', function(done) {
+      var idForListener;
+      client.then(t.on('create', function(key) {
+          console.log('[Event] Created key: ' + key);
+          throw new Error("Test error")
+      }))
+      .then(function(listenerId){
+          idForListener = listenerId;
+          return client;
+      })
+      .then(t.assert(t.putIfAbsent('exceptional-create', 'value'), t.toBeTruthy))
+      .then(function() {
+          done(new Error('Create event should throw an exception.'));
+      }, function(reason) {
+          expect(reason.message).toBe("Test error");
+          done();
+      }).finally(function() {
+          t.removeListenerFromClient(idForListener);
+      });
+  });
+  it('can fail when adding listener to disconnected client', function(done) {
+        var cli = t.client(t.local, {cacheName: 'namedCache'});
+        cli.then(function(client) {
+         client.disconnect();
+            return client;
+        }).then(function(client) {
+            client.addListener("create", function(key) {
+                console.log('[Event] Created new key1: ' + key);
+                throw new Error("This should not happen.");
+            });
+            return client;
+        }).then(function(client) {
+           client.put("testInput", "test");
+          return client;
+        }).then(function() {
+            done();
+        }, function(reason) {
+            console.log(reason);
+            done(new Error('The rejection is done silently. No exception should be thrown.'));
+        });
+    });
+
+  it('can reject removeListener call if the listenerId is invalid', function(done) {
+      client.then(t.removeListenerFromClient('fakeId')).then(function() {
+          done();
+      }, function(reason) {
```

Comment:
You don't need a `function() { done(); }`, you can just call `.then(done).catch(err) {})`...